sensu/transport/rabbitmq/rabbitmq.go

```diff
@@ -2,30 +2,84 @@ package rabbitmq
 
 import (
 	"errors"
-	"github.com/upfluence/sensu-go/Godeps/_workspace/src/github.com/upfluence/goutils/log"
+	"math/rand"
+	"time"
 
-	"github.com/upfluence/sensu-go/Godeps/_workspace/src/github.com/streadway/amqp"
+	"github.com/upfluence/sensu-client-go/Godeps/_workspace/src/github.com/upfluence/goutils/log"
+
+	"github.com/upfluence/sensu-client-go/Godeps/_workspace/src/github.com/streadway/amqp"
 )
 
 type RabbitMQTransport struct {
-	URI            string
 	Connection     *amqp.Connection
 	Channel        *amqp.Channel
 	ClosingChannel chan bool
+	Configs        []*TransportConfig
+}
+
+func NewRabbitMQTransport(uri string) (*RabbitMQTransport, error) {
+	config, err := NewTransportConfig(uri)
+
+	if err != nil {
+		log.Errorf("Received invalid URI %s", err)
+		return nil, err
+	}
+
+	return NewRabbitMQHATransport([]*TransportConfig{config}), nil
 }
 
-func NewRabbitMQTransport(uri string) *RabbitMQTransport {
-	return &RabbitMQTransport{URI: uri, ClosingChannel: make(chan bool)}
+func NewRabbitMQHATransport(configs []*TransportConfig) *RabbitMQTransport {
+	return &RabbitMQTransport{
+		ClosingChannel: make(chan bool),
+		Configs:        configs,
+	}
 }
 
 func (t *RabbitMQTransport) GetClosingChan() chan bool {
 	return t.ClosingChannel
 }
 
 func (t *RabbitMQTransport) Connect() error {
+	var uri string
 	var err error
+	randGenerator := rand.New(rand.NewSource(time.Now().UnixNano()))
+
+	for _, idx := range randGenerator.Perm(len(t.Configs)) {
+		config := t.Configs[idx]
+		uri = config.GetURI()
+
+		log.Noticef("Trying to connect to URI: %s", uri)
 
-	t.Connection, err = amqp.Dial(t.URI)
+		// TODO: Figure out how to specify the Prefetch value as well
+		// See amqp.Channel.Qos (it doesn't seem to be used currently)
+
+		// TODO: Add SSL support via amqp.DialTLS
+
+		heartbeatString := config.Heartbeat.String()
+		if heartbeatString != "" {
+			var heartbeat time.Duration
+			heartbeat, err = time.ParseDuration(heartbeatString)
```

Comment:
use the type inference  assignation: `heartbeat, err := time.ParseDuration(heartbeatString)`